In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import re
import scipy.stats

In [5]:
data = pd.read_csv('athleticnet_players.csv')
zcruit = pd.read_csv("ZCruit_Cleaned_Data.csv")

C:\Users\AlvinYu\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3165: DtypeWarning: Columns (15,17,19,21,23,25,27,29,31,33) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [12]:
print("There are {} players with T&F data".format(data.shape[0]))
print("{}% of players in ZCruit database have T&F data".format(round(100 * (data.shape[0]/zcruit.shape[0])), 2))

There are 7672 players with T&F data
11% of players in ZCruit database have T&F data


In [13]:
data.head(5)

Zcruit ID       Player Name  Grad Year  \
0  7b8c6df6-b548-4ed4-ae9b-4a741d7746c6  Bryson Lightfoot       2021   
1  45be7538-29ad-4e4a-9d2e-be05b0d3f595       Jalik Lewis       2022   
2  ae1749c5-a0f1-4fbc-852b-1ac51f5b1404      Jaiden Brown       2022   
3  eb32fcc6-a8f5-41df-b524-a4fc824a14f3       Mark Phifer       2021   
4  9712f1c5-6319-49ba-b8df-70b4102d3737    Gerimiah Brown       2021   

                     High School Name High School City High School State  \
0               Birdville High School      Haltom City                TX   
1          Denison Senior High School          Denison                TX   
2  Southern Regional 9-10 High School       Manahawkin                NJ   
3            Santa Monica High School     Santa Monica                CA   
4                    Vaca High School        Vacaville                CA   

  4 x 100 Meter                                4 x 100 Meter Event  \
0        43.29s               UIL 5A District 07 (W-Th) Track Meet   
1        44.42s  Liberty Christian Bubba Joiner Invitational (F...   
2           NaN                                                NaN   
3        43.58s               Mira Costa vs Santa Monica Dual Meet   
4        45.02s                  Stocking Super Seven Invitational   

  4 x 100 Meter Occurred on 4 x 200 Meter  ... Javelin Occurred on Discus  \
0                  4/3/2019        98.08s  ...                 NaN    NaN   
1                  3/5/2020        96.86s  ...                 NaN    NaN   
2                       NaN           NaN  ...                 NaN    NaN   
3                 2/27/2020        96.31s  ...                 NaN    NaN   
4                 2/29/2020        97.20s  ...                 NaN    NaN   

  Discus Event Discus Occurred on Pole Vault Pole Vault Event  \
0          NaN                NaN        NaN              NaN   
1          NaN                NaN        NaN              NaN   
2          NaN                NaN        NaN              NaN   
3          NaN                NaN        NaN              NaN   
4          NaN                NaN        NaN              NaN   

  Pole Vault Occurred on Hammer Hammer Event Hammer Occurred on  
0                    NaN    NaN          NaN                NaN  
1                    NaN    NaN          NaN                NaN  
2                    NaN    NaN          NaN                NaN  
3                    NaN    NaN          NaN                NaN  
4                    NaN    NaN          NaN                NaN  

[5 rows x 54 columns]

In [18]:
tf_metrics = ["100 Meter", "200 Meter", "400 Meter", 
              "110 Meter Hurdles", "300 Meter Hurdles", 
              "High Jump", "Long Jump", "Triple Jump",
              "Shotput", "Discus", "Hammer"]
tf_metrics_w_player_info = ["Zcruit ID", "Player Name"] + tf_metrics

In [19]:
tf_data = data[tf_metrics_w_player_info]

In [20]:
tf_data

Zcruit ID         Player Name 100 Meter  \
0     7b8c6df6-b548-4ed4-ae9b-4a741d7746c6    Bryson Lightfoot    11.25s   
1     45be7538-29ad-4e4a-9d2e-be05b0d3f595         Jalik Lewis       NaN   
2     ae1749c5-a0f1-4fbc-852b-1ac51f5b1404        Jaiden Brown       NaN   
3     eb32fcc6-a8f5-41df-b524-a4fc824a14f3         Mark Phifer   12.51 s   
4     9712f1c5-6319-49ba-b8df-70b4102d3737      Gerimiah Brown    11.74s   
...                                    ...                 ...       ...   
7667  f1873c0b-a931-4604-a33c-4439e32789eb        Jerod Thomas    12.26s   
7668  1f7b327a-510e-4070-a3cd-bc0d2433010f       Tyler Bradley    11.51s   
7669  6bcf9f7b-4c2d-4db2-a19e-bdea9c358d04  Jesse James Garcia    12.23s   
7670  fb194f5e-bc21-4852-838b-443eebac9aaf     Isaiah Robinson    11.53s   
7671  167feedc-df71-41cd-a151-d34f8b20decf       Jordan Garcia    12.15s   

     200 Meter 400 Meter 110 Meter Hurdles 300 Meter Hurdles High Jump  \
0          NaN       NaN               NaN               NaN       NaN   
1       24.95s    60.38s               NaN               NaN     5' 6"   
2       25.90s       NaN               NaN               NaN       NaN   
3       24.94s    64.33s               NaN               NaN       NaN   
4      25.77 s       NaN               NaN            47.84s       NaN   
...        ...       ...               ...               ...       ...   
7667       NaN       NaN               NaN               NaN     5' 4"   
7668   23.05 s       NaN               NaN               NaN       NaN   
7669    24.16s       NaN               NaN               NaN       NaN   
7670    24.10s       NaN               NaN               NaN       NaN   
7671    26.05s       NaN               NaN               NaN     5' 4"   

      Long Jump Triple Jump    Shotput Discus Hammer  
0           NaN         NaN        NaN    NaN    NaN  
1        18' 1"         NaN        NaN    NaN    NaN  
2           NaN         NaN  40' 10.5"    NaN    NaN  
3           NaN         NaN        NaN    NaN    NaN  
4      17' 3.5"         NaN        NaN    NaN    NaN  
...         ...         ...        ...    ...    ...  
7667        NaN         NaN        NaN    NaN    NaN  
7668  16' 7.25"    36' 1.5"        NaN    NaN    NaN  
7669   14' 8.5"         NaN        NaN    NaN    NaN  
7670     19' 6"         NaN        NaN    NaN    NaN  
7671     16' 9"         NaN        NaN    NaN    NaN  

[7672 rows x 13 columns]

In [21]:
def nan_counter(data, metrics):
    nan_dict, readable_dict = {}, {}
    for metric in metrics:
        nan_dict[metric] = data[metric].isna().sum() / data.shape[0]
        readable_dict[metric] = "There are {} NaNs out of {} players".format(data[metric].isna().sum(), data.shape[0])
    print(readable_dict)
    return nan_dict

In [22]:
nan_counter(tf_data, tf_metrics)

{'100 Meter': 'There are 3544 NaNs out of 7672 players', '200 Meter': 'There are 3561 NaNs out of 7672 players', '400 Meter': 'There are 5652 NaNs out of 7672 players', '110 Meter Hurdles': 'There are 7064 NaNs out of 7672 players', '300 Meter Hurdles': 'There are 7042 NaNs out of 7672 players', 'High Jump': 'There are 6693 NaNs out of 7672 players', 'Long Jump': 'There are 5204 NaNs out of 7672 players', 'Triple Jump': 'There are 6634 NaNs out of 7672 players', 'Shotput': 'There are 5387 NaNs out of 7672 players', 'Discus': 'There are 5893 NaNs out of 7672 players', 'Hammer': 'There are 7665 NaNs out of 7672 players'}


{'100 Meter': 0.4619395203336809,
 '200 Meter': 0.464155370177268,
 '400 Meter': 0.7367049009384776,
 '110 Meter Hurdles': 0.9207507820646507,
 '300 Meter Hurdles': 0.9178832116788321,
 'High Jump': 0.872393117831074,
 'Long Jump': 0.678310740354536,
 'Triple Jump': 0.8647028154327424,
 'Shotput': 0.7021637122002086,
 'Discus': 0.7681178310740354,
 'Hammer': 0.9990875912408759}

We can throw out hammer since 99.9% of players don't have a Hammer throw.

In [32]:
pandas.Series.str.replace(' ', '')
"^(\d)'\s*(\d*\.*\d*)"
def height_str_to_val(height_str):
    match = re.match("^(\d)'\s*(\d*\.*\d*)", height_str)
    groups = match.groups()
    feet, inches = int(groups[0]), float(groups[1])
    return feet * 12 + inches 

ValueError: could not convert string to float: '25. 77'

In [39]:
tf_data['100 Meter'].str.rstrip('s(.0)(.1)(.2)(.3)(.4)(.5)(.6)(.7)(.8)(.9)').astype(float)

ValueError: could not convert string to float: ''

In [44]:
pd.set_option('display.max_rows', 200)
tf_data[tf_data['100 Meter'].notna()].head(200)

Zcruit ID         Player Name      100 Meter  \
0    7b8c6df6-b548-4ed4-ae9b-4a741d7746c6    Bryson Lightfoot         11.25s   
3    eb32fcc6-a8f5-41df-b524-a4fc824a14f3         Mark Phifer        12.51 s   
4    9712f1c5-6319-49ba-b8df-70b4102d3737      Gerimiah Brown         11.74s   
5    c98e0321-f3a4-4b85-ab2a-bbd16d46ea86         Isaac Jones         11.73s   
6    7e546367-e3bd-44b4-a1e7-1b2cf182d224        David Godsey         11.17s   
7    20396bae-d68b-4b18-858e-e341d20f2b85          Hunt Young         12.22s   
8    a6b7e3c4-832d-4f6d-a039-e2af26012e82      Jahbez Hawkins         11.24s   
10   3df5eead-7148-452e-a38d-8adf743013a1    D'Angelo Stocker         10.99s   
11   b5c9b532-5606-474b-ae57-3431fed280bf          Cole Joyce        11.35 s   
12   16002f03-bfe1-49d8-b7b8-7ab827c86399   Fitzroy Ledgister         11.64s   
13   e2936003-f909-4d01-9f48-836d664aaf7a     Alshadee Salaam         11.29s   
14   52031735-d3cc-48cd-ae8f-92c0b4659ba1        Jemel Carter        11.12 s   
15   a1feaa9b-baa2-43b7-baec-ec08ac0e6c17         Drew Gallen         11.22s   
17   1bb159dd-b94a-4684-8493-fde202510adc       Jordan Doucet         11.24s   
18   40efd2c2-389a-46d1-af39-065243eeb086    Markevus Jackson         11.71s   
21   41bcab75-fa5b-4f31-80a7-dd8225339aec        Antonio Sims        11.47 s   
23   f42d8823-89aa-4b8c-a16e-65967a736250            AJ Green         11.16s   
24   8188ae16-c24c-4e89-a39c-cff4b8a75f31          Cody Hagen        11.00 s   
25   d6382a2a-e373-4bc4-9641-352a446ae59f    Makari Slaughter        11.05 s   
26   e98400f7-6667-4297-b3a0-fc9ea0fe4453      Xavier Preston         11.61s   
27   c1d4cdc7-7ad8-4f12-af2d-78a6b3b658d6         Javin Burke         10.84s   
29   2115242c-bb5b-4ae6-b39d-484fd3a0c279    Harrison Taggart        11.21 s   
31   c349ade6-0535-4cba-b458-5c9f020ba1fa      Lukkas Londono         12.21s   
32   4e2afbc0-c06c-4fb7-81cc-d38af5ef4870    Tysean Baronette        11.36 s   
33   1af312f0-bc4f-47e5-9284-57561da9237c        Isaiah Smith         11.92s   
35   bc33fcdd-7a5f-4f26-949a-0ea5b4115cdb     Darian Chestnut         12.18s   
37   9a272986-010e-48d4-be33-5d098d2d819f        Jeremiah Lee         11.50s   
38   eaffa473-94c1-4afe-b159-a9cb849d4581       Kevin Boswell        11.52 s   
39   aa844d21-757a-4e91-a3bb-18f7fcf3d11b      Ethan Capitano         11.81s   
40   afb27a1a-2ea9-40f4-912d-79f7100b9c06        Darius Brown         11.92s   
41   2589aff3-8cc3-4e2e-9bd4-e6bcd29bf8ea       Sahmir Hagans        11.23 s   
43   bf39d595-d6c9-4745-b543-a812dad533e7         Kuno Mendez         12.00s   
45   ba53cc11-f23d-4d46-888e-7880905ec016       Jaden Roberts        11.81 s   
47   dcf3b988-5494-40e6-bf70-2f122bab209f     Kaelen Mitchell         11.90s   
48   9a7198f4-5020-4989-863d-b6ca612ddd7b    Joshua Thornhill         12.19s   
51   651f2d04-fcaf-40a9-924d-7c35112bb6a7           Noah Kjar        11.37 s   
52   ac5a3a09-1d56-40f4-b4e4-8a833135137e         Wesley Maze         11.45s   
53   4076cdbe-881b-44d6-af2f-6d4a588a7efb      Tavon Hardwick    11.34 (.0)s   
54   0500ad3b-c433-44b1-9677-62408409a968     Talmage Handley        12.03 s   
55   8385c3ff-e6be-43b7-b667-bae394a4ac84       Shemar Wilson         12.73s   
56   168aa427-e4d5-43f9-8773-f774657bd0fa         Alonzo Sims        695.24s   
57   ae312f4d-5fb0-4783-b833-80572f475974  Teddy Lockhart III        11.47 s   
59   0a469afe-9cdd-471e-9b68-90957cf8564e        Jayden Smith         11.75s   
60   e46d3dfd-1279-4501-af82-065f03958f33      Jadye Foxworth         12.02s   
61   6686ed26-6725-4c92-803b-5a6cea9e6d24      Andrew Johnson         12.26s   
64   07a2cdc2-176b-4bde-8d53-75bc65400c47       Jalen Phillip         12.12s   
65   993256a8-adca-4401-9250-47c1eeea21ed         Jaydin Boyd         13.62s   
66   1597cec6-0f4a-47d3-9947-38f01e2d31cb       Ryan Fletcher         11.53s   
67   cb739f3f-df8a-4600-a4c6-7af8c5645382         Kaleb Brown         12.31s   
69   cf5e7915-5604-4626-989c-1fd

In [ ]:
.